### PR TITLE
Remove FQCN type hints on properties

### LINKED ISSUE #44364: Typehinted properties on missing class with preloading
### Symfony version(s) affected

6.0

### Description

I'm using `Symfony\Bridge\Twig\Mime\BodyRenderer` without `League\HTMLToMarkdown\HtmlConverter`.
The code in the constructor allows me to do that.

But in production, with preloading, I'm facing an issue:

```
Failed to load class League\HTMLToMarkdown\HtmlConverter used by typed property Symfony\Bridge\Twig\Mime\BodyRenderer::$converter during preloading
```

I believe that typehinted property HAVE to be loadable when preloaded

edit: same for `Predis\ClientInterface` in `Symfony\Component\Cache\Traits\RedisClusterProxy`

### How to reproduce

- enable preloading
- `composer require symfony/twig-bridge:^6.0`
- assert `League\HTMLToMarkdown` is NOT installed

### Possible Solution

- remove type hinted property for optional dependencies

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | #44364
| License       | MIT
| Doc PR        | n/a

This is an alternative for #44380.
Basically, we are removing all property type hints on 6.0 and reverts that for 6.1.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php [hunk 1]
```diff
@@ -22,7 +22,7 @@
  */
 class DoctrineChoiceLoader extends AbstractChoiceLoader
 {
-    private ObjectManager $manager;
+    private $manager;
     private string $class;
     private ?IdReader $idReader;
```
Comment (fancyweb) [inline src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php]: Nullable properties are not concerned? :thinking:

File: src/Symfony/Bridge/Doctrine/IdGenerator/UuidGenerator.php [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 final class UuidGenerator extends AbstractIdGenerator
 {
-    private UuidFactory $protoFactory;
+    private $protoFactory;
     private UuidFactory|NameBasedUuidFactory|RandomBasedUuidFactory|TimeBasedUuidFactory $factory;
```
Comment (fancyweb) [inline src/Symfony/Bridge/Doctrine/IdGenerator/UuidGenerator.php]: Union types are not concerned? :thinking:

File: src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php]: all types are concerned

File: src/Symfony/Bridge/Doctrine/IdGenerator/UuidGenerator.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Bridge/Doctrine/IdGenerator/UuidGenerator.php]: yes they are

Comment (stof) [reply src/Symfony/Bridge/Doctrine/IdGenerator/UuidGenerator.php]: I guess the removal was done with a search and replace with a regex that was not allowing `?` or `|` when matching the type...

**Comment by fabpot** [comment]
This one is ready

**Review by nicolas-grekas — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Pull request merged**